OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -144,46 +153,53 @@ public void onClick(View v) {
             }
         });
 
-        setupRegistrationForm();
+        callServerToGetRegistrationForm();
+
         hideSoftKeypad();
         tryToSetUIInteraction(true);
     }
 
-    public void showAgreement(RegistrationAgreement agreement) {
-        boolean isInAppEULALink = false;
-        try {
-            Uri uri = Uri.parse(agreement.getLink());
-            if (uri.getScheme().equals("edxapp")
-                    && uri.getHost().equals("show_eula")) {
-                isInAppEULALink = true;
-            }
-        } catch (Exception ex) {
-            logger.error(ex);
+    public void callServerToGetRegistrationForm() {
+        if (!NetworkUtil.isConnected(this)) {
+            showAlertDialog(getString(R.string.no_connectivity),
```

Comment:
No need to show this dialog as this server call is not based on a user action. We should be using a full screen error instead.